supervisor/supervisor.go

```diff
@@ -48,9 +48,6 @@ func New(stateDir string, runtimeName, shimName string, runtimeArgs []string, ti
 	}
 	go s.exitHandler()
 	go s.oomHandler()
-	if err := s.restore(); err != nil {
```

Comment:
There's not point in registering the API if it's not usable, and it is not usable until restore has been successful